src/organizations/components/OrgProfileTab/DeletePaidOrgOverlay.tsx

```diff
@@ -0,0 +1,218 @@
+// Libraries
+import React, {FC, useContext, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {
+  Alert,
+  AlignItems,
+  Button,
+  ComponentColor,
+  ComponentSize,
+  ComponentStatus,
+  FlexBox,
+  FlexDirection,
+  IconFont,
+  Input,
+  InputLabel,
+  InputType,
+  Overlay,
+} from '@influxdata/clockface'
+
+// Selectors
+import {
+  selectCurrentAccount,
+  selectCurrentOrg,
+  selectUser,
+} from 'src/identity/selectors'
+
+// API
+import {deleteOrganization} from 'src/identity/apis/org'
+
+// Constants
+import {CLOUD_URL} from 'src/shared/constants'
+
+// Notifications
+import {
+  deleteOrgDelayed,
+  deleteOrgFailed,
+  deleteOrgSuccess,
+} from 'src/shared/copy/notifications'
+import {notify} from 'src/shared/actions/notifications'
+
+// Utils
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+
+// Styles
+import './DeletePaidOrgOverlay.scss'
+
+// Required to avoid unwarranted console errors from clockface <Input /> when type=radio.
+const noop = () => null
+
+const linkStyle = {
+  color: 'white',
+  textDecoration: 'underline',
+}
+
+const SupportLink = (): JSX.Element => {
+  return (
+    <a
+      data-testid="go-to-new-org--link"
+      href="mailto:support@influxdata.com"
+      style={linkStyle}
+      target="_blank"
+    >
+      support@influxdata.com
+    </a>
+  )
+}
+
+export const DeletePaidOrgOverlay: FC = () => {
+  const account = useSelector(selectCurrentAccount)
+  const org = useSelector(selectCurrentOrg)
+  const user = useSelector(selectUser)
+  const dispatch = useDispatch()
+  const {onClose} = useContext(OverlayContext)
+
+  const [deleteButtonStatus, setDeleteButtonStatus] = useState(
+    ComponentStatus.Disabled
+  )
+  const [userAcceptedTerms, setUserAcceptedTerms] = useState(false)
+
+  const orgDeleteInProgress = deleteButtonStatus === ComponentStatus.Loading
+  const onClickCancel = orgDeleteInProgress ? noop : onClose
+
+  const toggleAcceptedTerms = () => {
+    const currentAcceptanceStatus = !userAcceptedTerms
+    setUserAcceptedTerms(currentAcceptanceStatus)
+
+    setDeleteButtonStatus(
+      currentAcceptanceStatus
+        ? ComponentStatus.Default
+        : ComponentStatus.Disabled
+    )
+  }
+
+  const handleDeleteOrg = () => {
+    setDeleteButtonStatus(ComponentStatus.Loading)
+
+    const responseTimer = setTimeout(() => {
+      dispatch(notify(deleteOrgDelayed(SupportLink)))
+
+      toggleAcceptedTerms()
+    }, 5000)
+
+    deleteOrganization(org.id)
+      .then(() => {
+        clearTimeout(responseTimer)
+        dispatch(notify(deleteOrgSuccess(org.name, account.name)))
+        setTimeout(() => {
+          onClose()
+          window.location.href = CLOUD_URL
+        }, 4000)
```

Comment:
I think we should consider storing the value of these [magic numbers](https://stackoverflow.com/a/47902/92446 into variables.